### PR TITLE
fix mutable dht entries in python bindings

### PR DESCRIPTION
**Pull request by bit**

### CODE REVIEW

File: bindings/python/src/session.cpp [hunk 1]
```diff
@@ -683,9 +715,9 @@ void bind_session()
         .def("set_dht_settings", allow_threads(&lt::session::set_dht_settings))
         .def("get_dht_settings", allow_threads(&lt::session::get_dht_settings))
         .def("dht_get_immutable_item", allow_threads(dht_get_immutable_item))
-        .def("dht_get_mutable_item", allow_threads(dht_get_mutable_item))
+        .def("dht_get_mutable_item", &dht_get_mutable_item)
```
Comment (arvidn) [inline bindings/python/src/session.cpp]: why no allow_threads here?

**Comment by arvidn** [comment]
did you see this?

https://ci.appveyor.com/project/arvidn/libtorrent/build/1.0.997/job/1ofmlf20f5h1k2fk

It looks like the use of this function in the binding is not conditional on TORRENT_DISABLE_DHT

session.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) void __cdecl libtorrent::dht::sign_mutable_item(struct std::pair<char const *,int>,struct std::pair<char const *,int>,unsigned __int64,char const *,char const *,char *)" (__imp_?sign_mutable_item@dht@libtorrent@@YAXU?$pair@PBDH@std@@0_KPBD2PAD@Z) referenced in function __catch$?session_set_settings@?A0x74070105@@YAXAAVsession@libtorrent@@ABVdict@python@boost@@@Z$0
bin\677dc9d9f8e04483fe03fa82e5e2b876\libtorrent.pyd : fatal error LNK1120: 1 unresolved externals

File: bindings/python/src/session.cpp (hunk 1 above)
Comment (bit) [inline bindings/python/src/session.cpp]: somehow failed to compile, can try again

Comment (bit) [inline bindings/python/src/session.cpp]: libtorrent/bindings/python/src/gil.hpp:69:30: error: return-statement with a value, in function returning 'void' [-fpermissive]
         return (a0.*fn)(a1,a2);

allow_threads does not work for functions that dont return anyting

**Comment by bit** [comment]
> TORRENT_DISABLE_DHT

**Comment by bit** [comment]
last update should fix the issues with TORRENT_DISABLE_DHT

**Comment by arvidn** [comment]
looks good to me. can you think of a simple way to make sure this binding won't break in the future? there is a small py-unit test in bindings/python that's run on travis and appveyor right now. I'm not sure it's that easy to construct these types though.

**Comment by arvidn** [comment]
merge failed. github is fairly strict in its automatic merging. could you rebase your patch on top of master?

**Pull request merged**